client/src/components/ExpandedRowComponent.tsx

```diff
@@ -0,0 +1,128 @@
+import React, {MouseEvent} from 'react';
+import type {CodeReview} from '../lib/types';
+import {format} from 'date-fns';
+import {logError} from '../services/log';
+import {useState} from 'react';
+import {fetchData} from '../services/fetch';
+
+export const useExpandedRowComponent = ({onUpdate, user}: {onUpdate: CallableFunction; user?: {displayName: string}}) => {
+    const ExpandedRowComponent = ({data}: {data: CodeReview}) => {
+        const handleActionClick = ({currentTarget}: MouseEvent<HTMLButtonElement>) => {
+            const action = currentTarget.getAttribute('name');
+            const value = currentTarget.getAttribute('value');
+
+            fetchData(`/api/action/${action}/${value}`)
+                .then((result) => {
+                    result?.data && onUpdate && onUpdate({
+                        data: result.data,
+                        action,
+                    });
+                })
+                .catch((e) => {
+                    setErrorMessage('Error saving data, see console error log for details.');
+                    logError(`Error sending: /api/action/${action}/${value}`, e);
+                })
+        };
+        const ActionButton = ({label, reviewId}: {label: string; reviewId: number}) => {
+            const sanitizedName = label.toLowerCase().replace(' ', '-');
+
+            return (
+                <button
+                    id={`${sanitizedName}-${reviewId}`}
+                    name={sanitizedName}
+                    value={reviewId}
+                    onClick={handleActionClick}
+                >{label}</button>
+            );
+        };
+        const showClaim = !user || !data.reviewers?.includes(user?.displayName);
+        const showApprove = !user || !data.approvers?.includes(user?.displayName);
+        const showChange = !user || !data.requestChanges?.includes(user?.displayName);
+        const [editing, setEditing] = useState(false);
+        const [errorMessage, setErrorMessage] = useState(null as null | string);
+        const [saveClicked, setSaveClicked] = useState(false);
+        const dirtyData = {...data};
+
+        const handleSaveClicked = ({currentTarget}: MouseEvent<HTMLButtonElement>) => {
+            if (JSON.stringify(dirtyData) !== JSON.stringify(data)) {
```

Comment:
There shouldn't be a situation where these values don't match - this is like treating the props like state.

Reply:
See previous comment, this was to handle row edit: Note, jira ticket inline on the expanded row.  Once data updated, it will trigger the callback to save the data to db, and signal the onUpdate to allow data table update without refetching.  This is a single row update inline.